### DISCUSSION TITLE
Typescript Test getStaticPaths

### DISCUSSION THREAD

**Discussion by kaaax0815** [vercel/next.js#24555] [Help] [answered]
[Source Code](https://github.com/kaaax0815/geis-bus-schedule)
Tests in "\_\_tests\_\_". Project generated with 'create-next-app'
```ts
// [linie].tsx
import { GetStaticPaths, GetStaticProps } from 'next';
export const getStaticPaths: GetStaticPaths = async () => {
  const res = await fetch('https://api.npoint.io/5853be5c4d0d6999f9d4');
  const json = await res.json();
  const linien = json.map((linie: Url) => linie.id);
  const paths = linien.map((linie: string) => ({ params: { linie: linie.replace(/ /g, '—') } }));
  return {
    paths,
    fallback: false
  };
};
```
```ts
// {linie].test.tsx
import { getStaticPaths, getStaticProps } from '../pages/[linie]';
const params = { linie: '8304' };
await getStaticPaths({ params });
```
Error Message:
```
Argument of type '{ params: { linie: string; }; }' is not assignable to parameter of type 'GetStaticPathsContext'.
  Object literal may only specify known properties, and 'params' does not exist in type 'GetStaticPathsContext'.
```

What argument do I need to test this? Without TypeScript it works, because its a Type Error.

**Comment by bobarros** [comment, marked as answer]
You cannot use getStaticPaths like that:

```jsx
await getStaticPaths({ params });
```

If you pay attention to this line, this function has no input:
```jsx
export const getStaticPaths: GetStaticPaths = async () => {
````

**Reply by kaaax0815** [reply]
But If I dont supply an argument, I get the Error "An argument for 'context' was not provided." and "Expected 1 arguments, but got 0.". Because the "GetStaticPaths" Type wants and argument "context"
type definition:
```js
export type GetStaticPaths<P extends ParsedUrlQuery = ParsedUrlQuery> = (
  context: GetStaticPathsContext
) => Promise<GetStaticPathsResult<P>>
```

**Reply by bobarros** [reply]
Why are you trying to execute StaticPaths somewhere? 
For next to create them, you don't need to do anything. 
It's like getStaticProps. You don't need to execute inside your code.

Here is an example that I have with GetStaticPaths:

```jsx
export function getAllRoutes() {
  const routes = require("../components/artigos/routes/routes.json");

  return routes.map((route) => {
    return {
      params: {
        url: [route.categoria, route.url],
      },
    };
  });
}

export async function getStaticPaths() {
  const paths = await getAllRoutes();
  return {
    paths,
    fallback: false,
  };
}

export async function getStaticProps({ params }) {
  const artigoMount = require("../components/html/Artigos/artigoMount"); 

  return {
    props: {
      artigoMount
    },
  };
}
```

**Reply by kaaax0815** [reply]
I want to test it. My normal site works perfectly fine. But I want to test the function and want to get 100% code coverage

**Reply by bobarros** [reply]
I don't think I was clear enough. You can use it, just not the way you're trying it.

It's getStaticProps that gets the params, not getStaticPaths.

```jsx
export async function getStaticProps({ params }) {
````
You don't have:
```jsx
export const getStaticPaths: GetStaticPaths = async ({ params }) => {
````
You can't execute it and you probably don't need to. Remove this line.
```jsx
await getStaticPaths({ params });
```

**Reply by kaaax0815** [reply]
But now I get only a coverage of 81% because these lines arent tested. can i somehow mock the context?

**Reply by bobarros** [reply]
If you really feel you need to test with some param, try this:

```jsx
await getStaticPaths({ locales: ["en"] });
```

For me, the real test is the output 🤷

**Reply by kaaax0815** [reply]
I know its not a real test, because their isnt anything really to break

**Reply by bobarros** [reply]
Ok.
So, locales and defaultLocale belongs to getStaticPaths ctx. Typescript is not going to complain.

**Reply by kaaax0815** [reply]
Thank u very much
